source/MaterialXGenMdl/MdlShaderGenerator.cpp

```diff
@@ -401,6 +522,53 @@ ShaderPtr MdlShaderGenerator::createShader(const string& name, ElementPtr elemen
         outputs->add(outputSocket->getSelf());
     }
 
+    // MDL does not allow varying data connected to transmission IOR.
+    // We must find all uses of transmission IOR and make sure we don't
+    // have a varying connection to it. If a varying connection is found
+    // we break that connection and revert to using default value on that
+    // instance of IOR, so that other uses of the same varying input still
+    // works in other places.
+    // As a result if a varying connections is set on transmission IOR 
+    // it just reverts to default value. Varying data on transmission IOR
+    // is very rare so this is normally not a problem in practice.
+    // One use-case where this fix is important is for shading models with
+    // a single IOR input, that gets connected to both reflection and 
+    // transmission IOR inside the shading model graph. For such cases
+    // this fix will disconnect the transmission IOR on the inside, but
+    // still support the connection to reflection IOR.
+    //
+    if (graph->hasClassification(ShaderNode::Classification::SHADER) ||
+        graph->hasClassification(ShaderNode::Classification::CLOSURE))
+    {
+        // Find dependencies on transmission IOR.
+        std::set<ShaderGraph*> graphsWithIorUniform;
+        std::set<ShaderGraph*> graphsWithIorVarying;
+        checkTransmissionIorDependencies(graph.get(), graphsWithIorUniform, graphsWithIorVarying);
+
+        // For any graphs found that has a varying connection
+        // to transmission IOR we break that connection.
+        for (ShaderGraph* g : graphsWithIorVarying)
+        {
+            disconnectTransmissionIor(g);
+            graphsWithIorUniform.erase(g);
```

Comment:
I'm not I understand this a graph can be classified as having uniform and varying connections at the same time
so once it's handled here you don't want to revisit it ?

Reply:
I understand the confusion.. I will clarify this better.